zipline/finance/performance/position_tracker.py

```diff
@@ -104,33 +152,71 @@ def position_values(self):
         self._position_last_sale_prices is changed.
         """
         if self._position_values is None:
-            vals = list(map(mul, self._position_amounts.values(),
```

Comment:
We're constructing and destroying a lot of potentially long lists here.

I would write this whole block as something like:
```
from six import itervalues
from itertools import izip
iter_amount_price_multiplier = izip(
    itervalues(self._position_amounts),
    itervalues(self._position_last_sale_prices),
    itervalues(self._position_value_multipliers),
)
self._position_values = [
    price * amount * multiplier for
    price, amount, multiplier in iter_amount_price_multiplier
]
```
cc @ehebert do we want to do this pre-emptively as part of your memory profiling work?